### DISCUSSION TITLE
React Nativeの基本/Stackのソースコードでコンパイルエラー

### DISCUSSION THREAD

**Discussion by nagano-ryo** [ws-4020/mobile-app-crib-notes#312] [General]
対象ページ
https://ws-4020.github.io/mobile-app-crib-notes/react-native/learn/basic-concepts/react-navigation-basics/stack

コードを張り付けたらコンパイルエラーが発生。
環境のせい？
アプリは動作する。

![image](https://user-images.githubusercontent.com/84065551/122508343-cebc6380-d03c-11eb-8144-d21e64dad196.png)

**Comment by nagano-ryo** [comment]
https://github.com/react-navigation/react-navigation/issues/9037
を参考に以下のように修正したらコンパイルエラーは解消した。

useNavigation()のジェネリクスにStackNavigationPropを指定。
```
// import {
//   createStackNavigator,
// } from '@react-navigation/stack';
import {
  createStackNavigator, StackNavigationProp,
} from '@react-navigation/stack';

// const navigation = useNavigation();
const navigation = useNavigation<StackNavigationProp<any>>();
```

**Comment by nagano-ryo** [comment]
以下にて対応済み。
https://github.com/ws-4020/mobile-app-crib-notes/pull/322

**Comment by creature-water-valley** [comment]
対応済みなのでクローズ
